File: taloa-web/types/tag.ts
export type TagStatus = "inactive" | "active" | "lost" | "disabled";

export interface PublicPet {
  name: string;
  species: string;
  breed_or_morph: string | null;
  sex: string | null;
  age_years: number | null;
  colour: string | null;
  photo_url: string | null;
}

export interface PublicProfile {
  allergies: string | null;
  medication: string | null;
  behaviour: string | null;
  likes: string | null;
  dislikes: string | null;
  public_notes: string | null;
  emergency_notes: string | null;
  vet_name: string | null;
  vet_phone: string | null;
}

export interface PublicContact {
  show_phone: boolean;
  show_email: boolean;
  phone: string | null;
  whatsapp: string | null;
  email: string | null;
}

export interface LostInfo {
  last_seen_at: string | null;
  last_seen_area: string | null;
  description: string | null;
}

export interface PublicTag {
  tag_code: string;
  status: TagStatus;
  tag_type: string | null;
  pet: PublicPet | null;
  profile: PublicProfile | null;
  contact: PublicContact | null;
  lost: LostInfo | null;
}

File: taloa-api/app/schemas/tag.py
"""Schemas da pagina publica da tag. NUNCA inclui dados privados do dono."""
from pydantic import BaseModel


class PublicPet(BaseModel):
    name: str
    species: str
    breed_or_morph: str | None = None
    sex: str | None = None
    age_years: int | None = None
    colour: str | None = None
    photo_url: str | None = None


class PublicProfile(BaseModel):
    # Apenas campos publicos — private_notes NUNCA entra aqui
    allergies: str | None = None
    medication: str | None = None
    behaviour: str | None = None
    likes: str | None = None
    dislikes: str | None = None
    public_notes: str | None = None
    emergency_notes: str | None = None
    vet_name: str | None = None
    vet_phone: str | None = None


class PublicContact(BaseModel):
    show_phone: bool = False
    show_email: bool = False
    phone: str | None = None       # so preenchido se show_phone = true
    whatsapp: str | None = None    # so digitos, para link wa.me
    email: str | None = None       # so preenchido se show_email = true


class LostInfo(BaseModel):
    last_seen_at: str | None = None
    last_seen_area: str | None = None
    description: str | None = None


class PublicTagResponse(BaseModel):
    tag_code: str
    status: str
    tag_type: str | None = None
    pet: PublicPet | None = None
    profile: PublicProfile | None = None
    contact: PublicContact | None = None
    lost: LostInfo | None = None
